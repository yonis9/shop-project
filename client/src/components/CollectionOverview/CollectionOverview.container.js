import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';

import { isFtechingSelector } from '../../redux/shop/shop-selectors';


import WithSpinner from '../WithSpinner/WithSpinner';
import CollectionOverview from './CollectionOverview';

const mapStateToProps = createStructuredSelector({
    isLoading: isFtechingSelector
})

const CollectionOverviewContainer = compose(
    connect(mapStateToProps),
    WithSpinner
)(CollectionOverview);

export default CollectionOverviewContainer;


