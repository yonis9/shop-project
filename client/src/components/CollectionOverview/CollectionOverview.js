import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { previewCollectionSelector } from '../../redux/shop/shop-selectors';

import CollectionPreview from '../CollectionPreview/CollectionPreview';

import { CollectionOverviewContainer } from './CollectionOverview.styles';

export const CollectionOverview = ( { collections }) => (
    <CollectionOverviewContainer>
         {
             collections.map(({ id, ...otherCollectionProps }) => (
                  <CollectionPreview key={id} {...otherCollectionProps}/>
               ))
          }
    </CollectionOverviewContainer>
)

const mapStateToProps = createStructuredSelector({
    collections: previewCollectionSelector
})


export default connect(mapStateToProps)(CollectionOverview);