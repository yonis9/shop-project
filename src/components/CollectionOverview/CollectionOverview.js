import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { previewCollectionSelector } from '../../redux/shop/shop-selectors';

import CollectionPreview from '../CollectionPreview/CollectionPreview';

import './CollectionOverview.scss';

const CollectionOverview = ( { collections }) => (
    <div className='collection-overview'>
         {
             collections.map(({ id, ...otherCollectionProps }) => (
                  <CollectionPreview key={id} {...otherCollectionProps}/>
               ))
          }
    </div>
)

const mapStateToProps = createStructuredSelector({
    collections: previewCollectionSelector
})


export default connect(mapStateToProps)(CollectionOverview);