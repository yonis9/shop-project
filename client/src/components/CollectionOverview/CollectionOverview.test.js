import React from 'react';
import { shallow } from 'enzyme';

import { CollectionOverview } from './CollectionOverview';

const wrapper = shallow(<CollectionOverview collections={[]}/>);

it('should rendered CollectionOverview component', () => {
    expect(wrapper).toMatchSnapshot();
});

