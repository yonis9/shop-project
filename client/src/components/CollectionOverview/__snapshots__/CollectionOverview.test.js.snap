// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendered CollectionOverview component 1`] = `ShallowWrapper {}`;
